examples/tv-casting-app/tv-casting-common/include/MediaCommandBase.h

```diff
@@ -29,10 +29,14 @@ class MediaCommandBase : public MediaBase
 
     CHIP_ERROR Invoke(RequestType request, std::function<void(CHIP_ERROR)> responseCallback)
     {
-        VerifyOrDieWithMsg(mTargetVideoPlayerInfo != nullptr, AppServer, "Target unknown");
+        ReturnErrorCodeIf(mTargetVideoPlayerInfo == nullptr, CHIP_ERROR_PEER_NODE_NOT_FOUND);
 
         auto deviceProxy = mTargetVideoPlayerInfo->GetOperationalDeviceProxy();
         ReturnErrorCodeIf(deviceProxy == nullptr || !deviceProxy->ConnectionReady(), CHIP_ERROR_PEER_NODE_NOT_FOUND);
+        ReturnErrorCodeIf(!deviceProxy->GetSecureSession().HasValue(), CHIP_ERROR_MISSING_SECURE_SESSION);
```

Comment:
The right answer here is to not hold on to operational device proxies.  They are supposed to be ephemeral objects, which you only use while under the "we got our CASE session" callback.  Any code that is storing them long term is going to randomly break when sessions get evicted.

mTargetVideoPlayerInfo should have a method that lets you pass in a callback function which will get the exchange manager and session and do the MediaClusterBase bits.  That method then should be talking to the CASESessionManager to set up a CASE session as needed, etc.

Reply:
And in particular, returning an error here is non-actionable, no?  How is a user supposed to deal with that error?